### ISSUE TITLE
No colorization with custom vterm invocation

### ISSUE THREAD

**Issue by tsdh** [akermu/emacs-libvterm#462]
Using `M-x vterm` works just fine and pops up a vterm buffer running my shell (fish) where colorization and everything else is indistinguishable from running fish in a standalone terminal emulator.

![image](https://user-images.githubusercontent.com/103854/101335808-f5bdff80-3879-11eb-995e-0012456d2291.png)

Awesome!

Now I've written a simple command I've bound to `F12` which toggles a vterm buffer popup where the CWD is the directory of the current file I'm editing.

```elisp
(defvar th/vterm-popup-buffer nil)

(defun th/vterm-popup ()
  "Toggle an vterm popup buffer with the current file's directory as cwd."
  (interactive)
  (let ((name " *Popup VTerm*"))
    (unless (buffer-live-p th/vterm-popup-buffer)
      (with-current-buffer (get-buffer-create name)
        (vterm-mode)
        (local-unset-key (kbd "F12"))
        (local-unset-key (kbd "<f12>")))
      (setq th/vterm-popup-buffer (get-buffer name)))
    (let ((win (get-buffer-window th/vterm-popup-buffer))
          (dir (file-name-directory (or (buffer-file-name)
                                        ;; in dired, there is no buffer-file-name
                                        dired-directory
                                        ;; VC buffers set this
                                        default-directory
                                        ;; use HOME
                                        "~/"))))
      (if win
          (quit-window nil win)
        (pop-to-buffer th/vterm-popup-buffer nil t)
        (unless (string= dir default-directory)
          (vterm-send-string (concat "cd " dir))
          (vterm-send-return))))))
```

Basically, that works as expected but for some reason, there is no colorization at all as lhown on the following screenshot:

![image](https://user-images.githubusercontent.com/103854/101335972-29992500-387a-11eb-9e1e-b4ba998035f3.png)

What's the reason for that?  If I look at the `vterm` command, I don't see anything initialization-wise that it would make different from what I do in my `th/vterm-popup` command.

**Comment by Sbozzolo** [comment]
I am not sure what the problem might be.

Does it work if you consider the minimal working kernel:
```emacs-lisp
(defun test-vterm ()
  (interactive)
  (with-current-buffer (get-buffer-create "TEST")
    (vterm-mode))
  (pop-to-buffer "TEST"))
```

?

**Comment by tsdh** [comment]
@Sbozzolo Yes, that one works.  And now I found the difference, too!  The colorization is somehow disabled when the buffer is hidden, i.e., the buffer's name starts with a space.  Here's a test case:

```elisp
(defun test-vterm ()
  (interactive)
  (require 'vterm)
  (with-current-buffer (get-buffer-create " TEST")
    (vterm-mode))
  (pop-to-buffer " TEST"))
```
I don't really see why the buffer name has an influence here...

**Comment by Sbozzolo** [comment]
We use `font-lock` to color the buffer. It sounds like that `font-lock` refuses to color hidden buffers (see, https://github.com/mooz/js2-mode/issues/313).

**Comment by tsdh** [comment]
Very interesting.  I've seen you put text properties in `vterm-module.c` and those are actually there.  It just happens that redisplay seems to ignore them in a hidden buffer.  If I copy a page from my hidden, uncolorized ` *Popup VTerm*` buffer and yank it in some other non-hidden buffer, the colors instantly appear as you've added them.
